### PR TITLE
add initial component infra docs

### PR DESCRIPTION
**Pull request by elmiko**

### CODE REVIEW

**Comment by elmiko** [comment]
@rvanderp3 and @dmage ptal, want to make sure i've got this accurate.

File: docs/component-infrastructure/index.md [hunk 1]
```diff
@@ -1,3 +1,41 @@
...
+Some components are not strictly required for an OpenShift installation, but
+their integration will improve user's experiences when operating OpenShift on
+an infrastructure provider.
+
+## Image Registry
+
+OpenShift contains a component named the [Image Registry Operator][image-reg-oper]
+which is responsible for managing a singleton instance of the OpenShift registry.
+This operator will manage a container image registry for the OpenShift cluster
+which will act as a local image cache for ImageStreams, built images (from Builds
```
Comment (dmage) [inline docs/component-infrastructure/index.md:12]: While it's a cache for imported images via image streams, it provides persistent storage for locally built images.

File: docs/component-infrastructure/index.md [hunk 2]
```diff
@@ -1,3 +1,41 @@
...
+and BuildConfigs), and intermediate images produced during build processes.
+
+With no infrastructure provider integration, the operator will deploy an image
+registry that is supported by local, ephemeral, storage. This means that if the
+node containing the registy is rebooted, then the cached images will be lost.
```
Comment (dmage) [inline docs/component-infrastructure/index.md:17]: This is not a supported configuration. This is a default behavior for new platforms. It allows the image registry to be available by default on a new platform in early alpha builds. It must be changed before this platform become feature complete/GA.

File: docs/component-infrastructure/index.md [hunk 3]
```diff
@@ -1,3 +1,41 @@
...
+
+Infrastructure providers can improve performance for their users by integrating
+an infrastructure specific storage configuration for the image registry
+operator. With proper integration to a storage service, images cached in the
```
Comment (dmage) [inline docs/component-infrastructure/index.md:21]: It's not about cacheing. The image registry is a persistent storage for local images. Either the platform provide such storage, or the image registry must be `Removed` by default on that platform.

**Comment by lobziik** [comment]
Make sense to me, thanks Mike!
lgtm

**Pull request merged**